hack/swagger.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Get the absolute path of this file
+DIR="$( cd "$( dirname "$0"  )" && pwd  )"
+cd "${DIR}" || return
+
+# Here is details of the swagger binary we use.
+# $ swagger version
+# version: 0.19.0
+SWAGGER_VERSION=v0.19.0
+
+. ./env.sh
+
+# swagger::check_version checks the command and the version.
+swagger::check_version() {
+  local has_installed version
+
+  has_installed="$(command -v swagger || echo false)"
+  if [[ "${has_installed}" = "false" ]]; then
+    echo false
+    return
+  fi
+
+  version="$(swagger version | head -n 1 | cut -d " " -f 2)"
+  if [[ "${SWAGGER_VERSION}" != "${version}" ]]; then
+    echo false
+    return
+  fi
+  echo true
+}
+
+# swagger::install installs the swagger binary.
+swagger::install() {
+    has_installed="$(swagger::check_version)"
+    if [[ "${has_installed}" = "true" ]]; then
+        echo ">>>> Detected that swagger-${SWAGGER_VERSION} has already installed. Skip installation. <<<<"
+        return
+    fi
+
+    echo ">>>> Start to install swagger-${SWAGGER_VERSION}. <<<<"
+    
+    local url
+    url="https://github.com/go-swagger/go-swagger/releases/download/${SWAGGER_VERSION}/swagger_${GOOS}_${GOARCH}"
+    wget --quiet -O /usr/local/bin/swagger "${url}" 
```

Comment:
I don't think it's a good idea to install an extra-tool in this directory. It may requires `root` permission, and make the user's operating system mess. If user has already installed `swagger` by himself, we just use it directly. If not, we can install them in our project's temporary directories: such as`.cache`, `bin` or others.